cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/utils/EmbeddedCassandraServerHelper.java

```diff
@@ -0,0 +1,356 @@
+package io.smartcat.cassandra.utils;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.commitlog.CommitLog;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.io.util.FileUtils;
+import org.apache.cassandra.service.CassandraDaemon;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.thrift.transport.TTransportException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.reader.UnicodeReader;
+
+import java.io.*;
+import java.net.ServerSocket;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+/**
+ * @author Jeremy Sevellec
+ */
+public class EmbeddedCassandraServerHelper {
```

Comment:
How can we pack up this in some mini project instead of copy pasting? We have commons and this will be same for each module which needs integration testing right?

We can put this along with yaml for configuration and log4j properties.